shaders/src/getters.vs

```diff
@@ -78,17 +78,18 @@ void skinPosition(inout vec3 p, const uvec4 ids, const vec4 weights) {
 void morphPosition(inout vec4 p) {
     ivec3 texcoord = ivec3(getVertexIndex() % MAX_MORPH_TARGET_BUFFER_WIDTH, getVertexIndex() / MAX_MORPH_TARGET_BUFFER_WIDTH, 0);
     for (uint i = 0u; i < objectUniforms.morphTargetCount; ++i) {
-        texcoord.z = int(i) * 2 + 0;
-        p += morphingUniforms.weights[i] * texelFetch(morphing_targets, texcoord, 0);
+        texcoord.z = int(i);
+        p += morphingUniforms.weights[i] * texelFetch(morphTargetBuffer_positions, texcoord, 0);
     }
 }
 
 void morphNormal(inout vec3 n) {
     ivec3 texcoord = ivec3(getVertexIndex() % MAX_MORPH_TARGET_BUFFER_WIDTH, getVertexIndex() / MAX_MORPH_TARGET_BUFFER_WIDTH, 0);
     for (uint i = 0u; i < objectUniforms.morphTargetCount; ++i) {
-        texcoord.z = int(i) * 2 + 1;
+        texcoord.z = int(i);
+        vec4 tangent = normalize(vec4(texelFetch(morphTargetBuffer_tangents, texcoord, 0)));
```

Comment:
The `normalize()`  is tricky, and I think this deserves at least a comment and possibly some precision adjustment.

The data in the texture is 16-bit signed integer, meaning texelFetch() will return a *float* value between -32768.0f and 32767.0f, and that doesn't fit in `mediump`, so the sampler might need to actually be `highp`.

What should happen, in theory, is a multiply by `1.0f / 32767.0f` and normalize shouldn't be needed (because the data is supposed to be mormalized AFAIK).

I *think* it's better to do the multiply by `1.0f / 32767.0f` to avoid overflow is normalize(), but  maybe we can live without the normalize() at all?

Reply:
Not only that but `normalize()` is much more expensive too.

Reply:
@pixelflinger is right. `unpackHalf2x16` is improper because packing way is different. I have changed sampler precision and devide by `32767.0f`